src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/InconsistentRoadClassificationCheck.java

```diff
@@ -112,7 +114,7 @@ public boolean validCheckForObject(final AtlasObject object)
                 .findInconsistentEdges(edge);
         if (!inconsistentEdgeTuples.isEmpty())
         {
-            final CheckFlag flag = createFlag(item,
+            final CheckFlag flag = createFlag(new OsmWayWalker((Edge) item).collectEdges(),
```

Comment:
The default OsmWayWalker (which is being used here) picks up reverse and master edges, so we'd want to change the ```nextCandidates``` function in the constructor.

Also, I'm not sure if this is the best place to use the WayWalker. Maybe it should be used in a larger sense to traverse the edge network looking for inconsistencies.

Reply:
I understand your concern about adding OSMWayWalker. I added it because the bug description mentioned about it. We can take input from Bentley and fix it according to that.